deep_qa/layers/tuple_matchers/embedded_tuple_matcher.py

```diff
@@ -18,17 +18,14 @@ class EmbeddedTupleMatcher:
         A reference to the TextTrainer object this TupleMatcher belongs to, so we can call
         ``embed_input`` with it.
 
-    tuple_matcher_params: Dict[str, Any], default={}
+    tuple_matcher_params: Dict[str, any], default={}
         Parameters for constructing an underlying TupleMatcher that operates on embedded tuples.
         We only read the "type" key here, which indexes a class in ``embedded_tuple_matchers``, and
         then pass the rest of the parameters on to that class as ``kwargs``.
     """
-    def __init__(self, text_trainer, tuple_matcher_params: Dict[str, Any]=None):
+    def __init__(self, text_trainer, tuple_matcher_params: Dict[str, Any]):
         self.text_trainer = text_trainer
-        if tuple_matcher_params is None:
-            tuple_matcher_params = {}
-        tuple_matcher_choice = get_choice_with_default(tuple_matcher_params,
-                                                       "embedded_matcher_type",
+        tuple_matcher_choice = pop_choice_with_default(tuple_matcher_params, "embedded_matcher_type",
```

Comment:
I think the only change necessary in this file is changing `get` to `pop`, and you can just revert the rest of them.  This doesn't currently allow a default value for `tuple_matcher_params`, and it would crash if you gave it `None`.